markdown.json

```diff
@@ -0,0 +1,62 @@
+{
+  "#": [
+    "Markdown is a shorthand for creating HTML from text strings."
+  ],
+  "methods": {
+    "description": [
+      "Check the public API is correct."
+    ],
+    "cases": [{
+      "description": "must be able to parse a Markdown string",
+      "method": "parse",
+      "arity": 1
+    }]
+  },
+  "cases": [
+    {
+       "description": "parses normal text as a paragraph",
+       "input": "This will be a paragraph",
+       "expected": "<p>This will be a paragraph</p>"
+    },
+    {
+      "description": "parsing italics",
+      "input": "_This will be italic_",
+      "expected": "<p><i>This will be italic</i></p>"
+    },
+    {
+      "description": "parsing bold text",
+      "input": "__This will be bold__",
+      "expected": "<p><em>This will be bold</em></p>"
+    },
+    {
+      "description": "mixed normal, italics and bold text",
+      "input": "This will _be_ __mixed__",
+      "expected": "<p>This will <i>be</i> <em>mixed</em></p>"
+    },
+    {
+      "description": "with h1 header level",
+      "input": "# This will be an h1",
+      "expected": "<h1>This will be an h1</h1>"
+    },
+    {
+      "description": "with h2 header level",
+      "input": "## This will be an h2",
+      "expected": "<h2>This will be an h2</h2>"
+    },
+    {
+      "description": "with h6 header level",
+      "input": "###### This will be an h6",
+      "expected": "<h6>This will be an h6</h6>"
+    },
+    {
+      "description": "unordered lists",
+      "input": "* Item 1\n* Item 2",
+      "expected": "<ul><li><p>Item 1</p></li><li><p>Item 2</p></li></ul>"
```

Comment:
To me it seems a bit inconsistent: There are `<p>` on the list items, but there are no `<p>` on list items in the next case ("With a little bit of everything"). This could plausibly be explained by the rule "If there are any tags on list items, don't add any, otherwise do add `<p>`" but it seems weird because I don't usually see `<p>` for list items.

What's more, I looked at http://parsedown.org/demo and https://daringfireball.net/projects/markdown/ and both do not emit `<p>` for the list items. I think they should be removed.

Reply:
As far as I understand the common mark spec, `<p>` should be there, if the list item contains more than a single paragraph:

http://spec.commonmark.org/0.28/#example-218 vs http://spec.commonmark.org/0.28/#example-219
----

218:
```md
- one

 two
```

```html
<ul>
<li>one</li>
</ul>
<p>two</p>
```

----

219
```md
- one

  two
```

```html
<ul>
<li>
<p>one</p>
<p>two</p>
</li>
</ul>
```